etna/search/forms.py

```diff
@@ -198,20 +237,238 @@ class BaseCollectionSearchForm(forms.Form):
         required=False,
     )
 
+    def _get_compressed_date_or_none_on_error(
```

Comment:
As much as possible, we should leave the implementation of the method up to the method, and not try to explain it in the name:
```suggestion
    def _get_compressed_date(
```